### ISSUE TITLE
Get epoch rewards for a block from node.

### ISSUE THREAD

**Issue by shaswatsaloni** [celo-org/celo-blockchain#2255]
Hi Team,

I have a rosetta as well as archival Celo Node. There i am getting the normal transactions with the query:

{
"method": "eth_getBlockByNumber",
"params": [
"0x556E00", //blocknumber
true
],
"id": 1,
"jsonrpc": "2.0"
}

But here the epoch transactions are not present, by which i want to have the voting rewards. 

But if i want to query the rosetta or archival node for this epoch transactions/rewards, how would i get that?

@eelanagaraj  could you please help us here.

Thanks,
Saloni

**Comment by carterqw2** [comment]
Hey @shaswatsaloni, could you please check out this [repo](https://github.com/celo-org/epochs) for an explainer on how epoch transactions work and how to fetch them.

**Comment by carterqw2** [comment]
Hi @shaswatsaloni! You're trying to get individual voter rewards, please refer to [this section](https://github.com/celo-org/epochs?tab=readme-ov-file#individual-voter-rewards) of the explainer for more details. Celo Explorer has to perform additional calculations to be able to display individual voter rewards that are not directly available through logs. I hope that helps.

**Comment by shaswatsaloni** [comment]
Okay, understood.

So, there is no simple way that we an fetch epoch transactions/ voting rewards for a celo address.

And, is there any way we can get the epoch transactions for an address instead of a block.

Thanks, 
Saloni

**Comment by carterqw2** [comment]
We provide an [RPC API](https://explorer.celo.org/mainnet/api-docs) in Celo Explorer, if that works for you (search for `getvoterrewards`).

https://explorer.celo.org/mainnet/api?module=epoch&action=getvoterrewards&voterAddress=0x9efecf85d7e668c0c3c7a778a1164b064234f96d

```
curl -X GET "https://explorer.celo.org/mainnet/api?module=epoch&action=getvoterrewards&voterAddress=0x9efecf85d7e668c0c3c7a778a1164b064234f96d" -H "accept: application/json"
```

**Comment by shaswatsaloni** [comment]
Yes, this i have. I just wanted to query my node if its at all possible?

**Comment by carterqw2** [comment]
Yeah, unfortunately, there is no easy way, you'd have to fetch epoch blocks and perform calculations [described](https://github.com/celo-org/epochs?tab=readme-ov-file#individual-voter-rewards) in the explainer manually.
